alibi_detect/cd/pytorch/context_aware.py

```diff
@@ -94,12 +95,7 @@ def __init__(
         self.meta.update({'backend': 'pytorch'})
 
         # set device
-        if device is None or device.lower() in ['gpu', 'cuda']:
-            self.device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
-            if self.device.type == 'cpu':
-                print('No GPU detected, fall back on CPU.')
-        else:
-            self.device = torch.device('cpu')
+        self.device = get_torch_device()
```

Comment:
This would have to be changed to:
```python
self.device = get_torch_device(device)
```
since `device` is a kwarg of the detector. Similar for the other detectors.